src/Instrumentation/Doctrine/_register.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+use OpenTelemetry\Contrib\Instrumentation\Doctrine\DoctrineInstrumentation;
+use OpenTelemetry\SDK\Sdk;
+
+if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(DoctrineInstrumentation::NAME) === true) {
+    return;
+}
+
+if (extension_loaded('opentelemetry') === false) {
```

Comment:
Is this necessary, considering you have ext-opentelemetry as a requirement in composer.json? If you want to allow project to still function without having the extension loaded, you should then also make it as optional dependency in composer.json.

Reply:
We've got this test in all auto-instrumentations, because folks will often install with composer's `--ignore-platform-reqs` (reported usages have been for CI pipelines, or a multi-developer project where not all developers care about auto-instrumentation/telemetry). We're open to revisiting this, but AFAIR those are the edge cases to consider.